test/distributed/_tensor/test_random_ops.py

```diff
@@ -26,6 +27,63 @@
 )
 
 
+class DistTensorRandomInitTest(DTensorTestBase):
+    def _run_init_op(self, init_op, *args, **kwargs):
+        device_mesh = self.build_device_mesh()
+        shard_spec = [Shard(0)]
+        input_size = (8, 4)
+
+        # NOTE: currently random initialization on cuda device has different
+        # behavior from other devices. Unify the test once the behavior is unified.
+        if not is_rng_supported_mesh(device_mesh):
+            input_tensor = torch.randn(*input_size, device=self.device_type)
+            dtensor = DTensor.from_local(input_tensor, device_mesh, shard_spec)
+            local_tensor_clone = torch.clone(input_tensor)
+            torch.manual_seed(self.rank)
+            local_tensor_clone = init_op(local_tensor_clone, *args, **kwargs)
+            torch.manual_seed(self.rank)
+            dtensor = init_op(dtensor, *args, **kwargs)
+            self.assertEqual(local_tensor_clone, dtensor.to_local())
+        else:
+            # initialize rng state
+            manual_seed(1234, device_mesh)
+
+            # create DTensor from Tensor
+            _tensor = torch.empty(*input_size, device="cuda")
+            dtensor = DTensor.from_local(_tensor, device_mesh, [Shard(1)])
+
+            # DTensor random init
+            dtensor = init_op(dtensor, *args, **kwargs)
+            local_tensor = dtensor.to_local()
+
+            # allgather the local tensors
+            dtensor = dtensor.redistribute(device_mesh, [Replicate()])
+            local_tensor_gathered = dtensor.to_local()
+
+            # compare with local tensors from other ranks
+            for other_rank in range(self.world_size):
+                if self.rank != other_rank:
+                    slice_idx = [
+                        slice(input_size[0]),
+                        slice(
+                            other_rank * input_size[1], (other_rank + 1) * input_size[1]
```

Comment:
what if `other_rank` is the last rank, would it still generate the right shape?

Reply:
Yes. I tried the following example:
```
>>> import torch
>>> tensor = torch.arange(32).reshape(4, 8)
>>> last_rank = 3
>>> slice_idx = [slice(4), slice(last_rank * 2, (last_rank + 1) * 2)]
>>> tensor[slice_idx]
tensor([[ 6,  7],
        [14, 15],
        [22, 23],
        [30, 31]])
```